route/route.go

```diff
@@ -330,6 +375,117 @@ func (r *Router) batch(w http.ResponseWriter, req *http.Request) {
 }
 
+func (r *Router) Export(ctx context.Context, req *collectortrace.ExportTraceServiceRequest) (*collectortrace.ExportTraceServiceResponse, error) {
+
+	md, ok := metadata.FromIncomingContext(ctx)
+	if !ok {
+		r.Logger.Error().Logf("Unable to retreive metadata from OTLP request.")
+		return &collectortrace.ExportTraceServiceResponse{}, nil
+	}
+
+	var requestID types.RequestIDContextKey
+
+	apiKey, dataset := getAPIKeyAndDatasetFromMetadata(md)
+	if apiKey == "" {
+		r.Logger.Error().Logf("Received OTLP request without Honeycomb APIKey header")
+		return &collectortrace.ExportTraceServiceResponse{}, nil
+	}
+	if dataset == "" {
+		r.Logger.Error().Logf("Received OTLP request without Honeycomb dataset header")
+		return &collectortrace.ExportTraceServiceResponse{}, nil
+	}
+
+	apiHost, err := r.Config.GetHoneycombAPI()
+	if err != nil {
+		r.Logger.Error().Logf("Unable to retrieve APIHost from config while processing OTLP batch")
+		return &collectortrace.ExportTraceServiceResponse{}, nil
+	}
+
+	for _, resourceSpan := range req.ResourceSpans {
+		resourceAttrs := make(map[string]interface{})
+
+		if resourceSpan.Resource != nil {
+			addAttributesToMap(resourceAttrs, resourceSpan.Resource.Attributes)
+		}
+
+		for _, librarySpan := range resourceSpan.InstrumentationLibrarySpans {
+			library := librarySpan.InstrumentationLibrary
+			if library != nil {
+				if library.Name != "" {
+					resourceAttrs["library.name"] = library.Name
+				}
+				if library.Version != "" {
+					resourceAttrs["library.version"] = library.Version
+				}
+			}
+
+			for _, span := range librarySpan.GetSpans() {
+				traceID := bytesToTraceID(span.TraceId)
+				spanID := hex.EncodeToString(span.SpanId)
+				timestamp := time.Unix(0, int64(span.StartTimeUnixNano)).UTC()
+
+				eventAttrs := map[string]interface{}{
+					"trace.trace_id": traceID,
+					"trace.span_id":  spanID,
+					"type":           getSpanKind(span.Kind),
+					"name":           span.Name,
+					"duration_ms":    float64(span.EndTimeUnixNano-span.StartTimeUnixNano) / float64(time.Millisecond),
+					"status_code":    int32(span.Status.Code),
+				}
+				if span.ParentSpanId != nil {
+					eventAttrs["trace.parent_id"] = hex.EncodeToString(span.ParentSpanId)
+				}
+				if r.getSpanStatusCode(span.Status) == trace.Status_STATUS_CODE_ERROR {
+					eventAttrs["error"] = true
+				}
+				if len(span.Status.Message) > 0 {
+					eventAttrs["status_message"] = span.Status.Message
+				}
+				if span.Attributes != nil {
+					addAttributesToMap(eventAttrs, span.Attributes)
+				}
+
+				sampleRate := 1
+				if eventAttrs["sampleRate"] != nil {
+					switch eventAttrs["sampleRate"].(type) {
+					case string:
+						sampleRate, err = strconv.Atoi(eventAttrs["sampleRate"].(string))
+						if err != nil {
+							r.Logger.Info().WithField("Unable to parse sampleRate in OTLP span attribute", eventAttrs["sampleRate"])
+						}
+					case int:
+						sampleRate = eventAttrs["sampleRate"].(int)
```

Comment:
Could add a `default:` here with a log statement stating that it couldn't parse the sampleRate (in case it's something other than a string or an int - say, a float).

Reply:
Added default case that creates err with info that goes in debug logs.